packages/server/src/app/db/notifications/notifications.service.ts

```diff
@@ -0,0 +1,38 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { CreateNotification, UnpublishedNotification } from '@pulp-fiction/models/notifications';
+import { Model } from 'mongoose';
+import { NotificationSubscriptionDocument } from './notification-subscriptions.schema';
+
+import { NotificationDocument } from './notifications.schema';
+import { UnpublishedNotificationDocument } from './unpublished-notifications.schema';
+
+@Injectable()
+export class NotificationsService {
+    constructor(@InjectModel('Notification') private readonly notificationModel: Model<NotificationDocument>,
+        @InjectModel('UnpublishedNotification') private readonly unpubNotifModel: Model<UnpublishedNotificationDocument>,
+        @InjectModel('NotificationSubscription') private readonly subscriptionModel: Model<NotificationSubscriptionDocument>) { }
+
+        async queueNotification(notification: CreateNotification): Promise<void> {
+            // Add it to the unpublished notification queue, where it'll (eventually)
+            // be processed.
+            await new this.unpubNotifModel(notification).save();
+        }
+
+        async getUnreadNotifications(userId: string): Promise<Notification[]> {
+            throw new Error("Not implemented yet");
+        }
+
+        async getAllNotifications(userId: string): Promise<Notification[]> {
+            throw new Error("Not implemented yet");
+        }
+
+        async subscribe(userId: string, sourceId: string): Promise<void> {
+            throw new Error("Not implemented yet");        
+        }
+
+        async unsubscribe(userId: string, sourceId: string): Promise<void> {
+            throw new Error("Not implemented yet");
```

Comment:
The nice thing here is that `subscribe()` and `unsubscribe()` can be super-simple-- we don't _care_ what kind of thing a user is subscribing to. It's the individual notifications' jobs to tell us what `Kind` of thing they are (which we can then use to determine what collections to go hunting in for that ID).